oscap build fails on Ubuntu 20.04 at oscap link step

#### Description of Problem:

post-cmake make of openscap fails at oscap linking step with error

```
/usr/bin/ld: ../src/libopenscap.so.25.2.0: undefined reference to `MMap::MMap(FileFd&, unsigned long)'
collect2: error: ld returned 1 exit status
make[2]: *** [utils/CMakeFiles/oscap.dir/build.make:239: utils/oscap] Error 1
make[1]: *** [CMakeFiles/Makefile2:3226: utils/CMakeFiles/oscap.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

#### OpenSCAP Version:

origin/maint-1.3 from https://github.com/OpenSCAP/openscap

#### Operating System & Version:

Ubuntu 20.04 LTS

#### Steps to Reproduce:

1. Follow Ubuntu-related directions in https://github.com/OpenSCAP/openscap/blob/maint-1.3/docs/developer/developer.adoc
2. Run cmake ../ and make
3. Receive ld link error


#### Actual Results:

Incomplete build

#### Expected Results:

Completed build

#### Additional Information / Debugging Steps:

Terminal history follows. Note that cmake identifies some absent potential components which may or may not be pertinent at this stage of build [oscap-1.3.3-on-ubuntu-20.04-build.log](https://github.com/OpenSCAP/openscap/files/4768828/oscap-1.3.3-on-ubuntu-20.04-build.log).

```
gapinski@nuc7i7bnh-650ncr-com:~/Projects/github/OpenSCAP/openscap/build$ git status
On branch maint-1.3
Your branch is up to date with 'origin/maint-1.3'.

nothing to commit, working tree clean
gapinski@nuc7i7bnh-650ncr-com:~/Projects/github/OpenSCAP/openscap/build$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"
gapinski@nuc7i7bnh-650ncr-com:~/Projects/github/OpenSCAP/openscap/build$ 

```

--- a/src/OVAL/probes/unix/linux/dpkginfo-helper.cxx
+++ b/src/OVAL/probes/unix/linux/dpkginfo-helper.cxx
@@ -15,33 +15,30 @@
 #include <apt-pkg/pkgcache.h>
 #include <apt-pkg/pkgrecords.h>
 #include <apt-pkg/pkgsystem.h>
+#include <apt-pkg/cachefile.h>
 
 #include "dpkginfo-helper.h"
 
 using namespace std;
 
 static int _init_done = 0;
-static pkgCache *cgCache = NULL;
+static pkgCacheFile *cgCache = NULL;
+
 static MMap *dpkg_mmap = NULL;
 
 static int opencache (void) {
         if (pkgInitConfig (*_config) == false) return 0;
-        if (pkgInitSystem (*_config, _system) == false) return 0;
 
         const char* root = getenv("OSCAP_PROBE_ROOT");
-        string pkgCacheRoot(root != NULL ? root : "");
-
-        FileFd *fd = new FileFd (pkgCacheRoot + _config->FindFile ("Dir::Cache::pkgcache"),
-                        FileFd::ReadOnly);
-
-        dpkg_mmap = new MMap (*fd, MMap::Public|MMap::ReadOnly);
-        if (_error->PendingError () == true) {
-                _error->DumpErrors ();
-                return 0;
+        if (root != NULL) {
+            string pkgCacheRoot(root);
+            _config->Set("RootDir", pkgCacheRoot);
         }
 
-        cgCache = new pkgCache (dpkg_mmap);
-        if (0 == cgCache) return 0;
+        if (pkgInitSystem (*_config, _system) == false) return 0;
+
+        if (!cgCache->ReadOnlyOpen(NULL)) return 0;
+
         if (_error->PendingError () == true) {
                 _error->DumpErrors ();
                 return 0;
@@ -52,7 +49,7 @@ static int opencache (void) {
 
 struct dpkginfo_reply_t * dpkginfo_get_by_name(const char *name, int *err)
 {
-        pkgCache &cache = *cgCache;
+        pkgCache &cache = *cgCache->GetPkgCache();
         pkgRecords Recs (cache);
         struct dpkginfo_reply_t *reply = NULL;
 
@@ -125,6 +122,7 @@ void dpkginfo_free_reply(struct dpkginfo_reply_t *reply)
 
 int dpkginfo_init()
 {
+        cgCache = new pkgCacheFile;
         if (_init_done == 0)
                 if (opencache() != 1) {
                         return -1;
@@ -135,6 +133,8 @@ int dpkginfo_init()
 
 int dpkginfo_fini()
 {
+        cgCache->Close();
+
         delete cgCache;
         cgCache = NULL;
 

